### PR TITLE
Documentation for endpoint gateway

### LINKED ISSUE #2995: docs for ibm_is_virtual_endpoint_gateway for COS do not exist
https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_virtual_endpoint_gateway

To get this to work for COS I believe is something like this:

```
resource "ibm_resource_instance" "cos" {
  count             = var.make_cos ? 1 : 0
  name              = "${var.basename}-cos"
  resource_group_id = var.resource_group_id
  service           = "cloud-object-storage"
  plan              = "standard"
  location          = "global"
  tags              = local.tags
}

resource "ibm_resource_key" "cos_key" {
  count                = var.make_cos ? 1 : 0
  name                 = "${var.basename}-cos-key"
  resource_instance_id = ibm_resource_instance.cos[0].id
  role                 = "Writer"

  parameters = {
    service-endpoints = "private"
  }
  tags = local.tags
}

resource "ibm_is_virtual_endpoint_gateway" "cos" {
  count          = var.make_cos ? 1 : 0
  vpc               = var.vpc.id
  name           = "${var.basename}-cos"
  resource_group = var.resource_group_id
  target {
    crn           = "crn:v1:bluemix:public:cloud-object-storage:global:::endpoint:s3.direct.${var.region}.cloud-object-storage.appdomain.cloud"
    resource_type = "provider_cloud_service"
  }
```

I could not figure this out from the docs provided.  Specifically the COS example in the ibm_is_virtual_endpoint_gateway.  An example like this needs to be added in the terraform ibm_is_virtual_endpoint_gateway docs.

```
resource "ibm_is_virtual_endpoint_gateway" "cos" {
  target {
    crn           = "crn:v1:bluemix:public:cloud-object-storage:global:::endpoint:s3.direct.${var.region}.cloud-object-storage.appdomain.cloud"
    resource_type = "provider_cloud_service"
  }
```

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #2995

Output from acceptance testing:

Screenshot 2021-08-26 at 5 01 40 PM

```
$ make testacc TEST=./ibm TESTARGS='-run=TestAccIBMISVirtualEndpointGateway_Basic'
```

### CODE REVIEW

File: website/docs/r/is_virtual_endpoint_gateway.html.markdown [hunk 1]
```diff
@@ -71,9 +81,9 @@ Review the argument references that you can specify for your resource.
 - `target` - (Required, List) The endpoint gateway target.
 
   Nested scheme for `target`:
-  - `crn` - (Optional, Forces new resource, String) The endpoint gateway target `CRN`. If CRN not specified, `name` must be specified. 
+  - `crn` - (Optional, Forces new resource, String) The CRN for this provider cloud service, or the CRN for the user's instance of a provider cloud service.. If CRN not specified, `name` must be specified. 
   - `name` - (Required, Forces new resource, String) The endpoint gateway target name.
-  - `resource_type` - (Required, String) The endpoint gateway target resource type.
+  - `resource_type` - (Required, String) The endpoint gateway target resource type, allowed values [ provider_cloud_service, provider_infrastructure_service ].
```
Comment (astha-jain) [inline website/docs/r/is_virtual_endpoint_gateway.html.markdown:86]: Can we use the following style to to show the enums and remove [ ] ?

The possible values are `cpu`, `memory`, `network_in` and `network_out`.

**Review by astha-jain — APPROVED** [review]
Added one comment, rest looks good

File: website/docs/r/is_virtual_endpoint_gateway.html.markdown (hunk 1 above)
Comment (SunithaGudisagarIBM) [reply website/docs/r/is_virtual_endpoint_gateway.html.markdown]: Incorporated this comment

File: ibm/resource_ibm_is_virtual_endpoint_gateway.go [hunk 2]
```diff
@@ -162,9 +162,10 @@ func resourceIBMISEndpointGateway() *schema.Resource {
 							Description: "The target crn",
 						},
 						isVirtualEndpointGatewayTargetResourceType: {
-							Type:        schema.TypeString,
-							Required:    true,
-							Description: "The target resource type",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateAllowedStringValue([]string{"provider_cloud_service", "provider_infrastructure_service"}),
```
Comment (hkantare) [inline ibm/resource_ibm_is_virtual_endpoint_gateway.go:167]: Can you please use InvokeValidator instead of validateAllowedStringValue

Comment (SunithaGudisagarIBM) [reply ibm/resource_ibm_is_virtual_endpoint_gateway.go]: made the required changes and tested as well
Screenshot 2021-09-01 at 4 35 58 PM

**Pull request merged**